src/spac/visualization.py

```diff
@@ -21,8 +21,10 @@
 
 
 def visualize_2D_scatter(
-    x, y, labels=None, point_size=None, theme=None,

```

Comment:
the source of the failures in the unittests are how you handle the parameter labels. the failing unittests are 

test_continuous_labels
test_labels_length_mismatch
test_legend_placement_categorical_labels

see the file https://github.com/FNLCR-DMAP/spac_datamine/blob/pin_colors/tests/test_visualization/test_visualize_2D_scatter.py